### PR TITLE
Test for issue #1339

### PR DESCRIPTION
**Pull request by Jurigag**
Test for issue #1339

### CODE REVIEW

**Comment by Jurigag** [comment]
@sjinks any idea why this works on clang and not gcc?

Ping me here if you would manage to fix something so i will pull changes.

File: unit-tests/ci/after_failure.sh [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 shopt -s nullglob
 export LC_ALL=C
-for i in core core.*; do
+VERSION=$(php-config --version)
+
+for i in /tmp/core_*.*; do
 	if [ -f "$i" -a "$(file "$i" | grep -o 'core file')" ]; then
```
Comment (sjinks) [inline unit-tests/ci/after_failure.sh:8]: This check probably needs to go away as well

Comment (Jurigag) [reply unit-tests/ci/after_failure.sh:8]: Well it's like this on phalcon repo and it's working well.

**Comment by sergeyklay** [comment]
As I can see clang and vc++ passed :)

**Comment by sjinks** [comment]
I was able to reproduce this :-) NTS build of PHP 7.0, and the extension must be compiled with gcc 4.8.x series

**Comment by sjinks** [comment]
```
Starting test 'Extension\InstanceTest::testIssue1339'.
==8777== Warning: client switching stacks?  SP change: 0xffeffbcc0 --> 0x133f4600
==8777==          to suppress, use: --max-stackframe=68379768512 or greater
==8777== Invalid read of size 8
==8777==    at 0x106801E3: zim_Test_Instance_testIssue1339 (instance.zep.c:150)
==8777==    by 0xFFFFFFFF00000C07: ???
==8777==    by 0xAC: ???
==8777==    by 0x1190FF3F: ???
==8777==    by 0x12C2D9F: ??? (in /home/vladimir/.phpenv/versions/7.1.0-nts-debug/bin/php)
==8777==  Address 0x133f4600 is 384 bytes inside a block of size 576 free'd
==8777==    at 0x4C2DD6B: free (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==8777==    by 0x86D399: _efree (zend_alloc.c:2428)
==8777==    by 0x8BA802: zend_array_destroy (zend_hash.c:1330)
==8777==    by 0x8A0B1D: _zval_dtor_func (zend_variables.c:43)
==8777==    by 0x889637: i_zval_ptr_dtor (zend_variables.h:48)
==8777==    by 0x88B21B: _zval_ptr_dtor (zend_execute_API.c:550)
==8777==    by 0x10606E7B: zephir_memory_restore_stack_common (memory.c:163)
==8777==    by 0x1060720A: zephir_memory_restore_stack (memory.c:244)
==8777==    by 0x106801E1: zim_Test_Instance_testIssue1339 (instance.zep.c:148)
==8777==  Block was alloc'd at
==8777==    at 0x4C2CB3F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==8777==    by 0x86E029: __zend_malloc (zend_alloc.c:2820)
==8777==    by 0x86D2F2: _emalloc (zend_alloc.c:2413)
==8777==    by 0x8B6A3A: zend_hash_real_init_ex (zend_hash.c:138)
==8777==    by 0x8B6DE5: zend_hash_real_init (zend_hash.c:201)
==8777==    by 0x1060D1E6: zephir_create_array (array.c:48)
==8777==    by 0x1067EDF3: zim_Test_Instance_testIssue1339 (instance.zep.c:68)
==8777==    by 0x908EF2: ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER (zend_vm_execute.h:1097)
==8777==    by 0x90728E: execute_ex (zend_vm_execute.h:429)
==8777==    by 0x88C4F7: zend_call_function (zend_execute_API.c:828)
==8777==    by 0x5F5C01: reflection_method_invoke (php_reflection.c:3316)
==8777==    by 0x5F5DD1: zim_reflection_method_invokeArgs (php_reflection.c:3352)
==8777==
==8777== Invalid read of size 8
==8777==    at 0x106801E4: zim_Test_Instance_testIssue1339 (instance.zep.c:150)
==8777==    by 0xFFFFFFFF00000C07: ???
==8777==    by 0xAC: ???
==8777==    by 0x1190FF3F: ???
==8777==    by 0x12C2D9F: ??? (in /home/vladimir/.phpenv/versions/7.1.0-nts-debug/bin/php)
==8777==  Address 0x133f4608 is 392 bytes inside a block of size 576 free'd
==8777==    at 0x4C2DD6B: free (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==8777==    by 0x86D399: _efree (zend_alloc.c:2428)
==8777==    by 0x8BA802: zend_array_destroy (zend_hash.c:1330)
==8777==    by 0x8A0B1D: _zval_dtor_func (zend_variables.c:43)
==8777==    by 0x889637: i_zval_ptr_dtor (zend_variables.h:48)
==8777==    by 0x88B21B: _zval_ptr_dtor (zend_execute_API.c:550)
==8777==    by 0x10606E7B: zephir_memory_restore_stack_common (memory.c:163)
==8777==    by 0x1060720A: zephir_memory_restore_stack (memory.c:244)
==8777==    by 0x106801E1: zim_Test_Instance_testIssue1339 (instance.zep.c:148)
==8777==  Block was alloc'd at
==8777==    at 0x4C2CB3F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==8777==    by 0x86E029: __zend_malloc (zend_alloc.c:2820)
==8777==    by 0x86D2F2: _emalloc (zend_alloc.c:2413)
==8777==    by 0x8B6A3A: zend_hash_real_init_ex (zend_hash.c:138)
==8777==    by 0x8B6DE5: zend_hash_real_init (zend_hash.c:201)
==8777==    by 0x1060D1E6: zephir_create_array (array.c:48)
==8777==    by 0x1067EDF3: zim_Test_Instance_testIssue1339 (instance.zep.c:68)
==8777==    by 0x908EF2: ZEND_DO_FCALL_SPEC_RETVAL_USED_HANDLER (zend_vm_execute.h:1097)
==8777==    by 0x90728E: execute_ex (zend_vm_execute.h:429)
==8777==    by 0x88C4F7: zend_call_function (zend_execute_API.c:828)
==8777==    by 0x5F5C01: reflection_method_invoke (php_reflection.c:3316)
==8777==    by 0x5F5DD1: zim_reflection_method_invokeArgs (php_reflection.c:3352)
==8777==
==8777== Jump to the invalid address stated on the next line
==8777==    at 0xFFFFFFFF00000C08: ???
==8777==    by 0xAC: ???
==8777==    by 0x1190FF3F: ???
==8777==    by 0x12C2D9F: ??? (in /home/vladimir/.phpenv/versions/7.1.0-nts-debug/bin/php)
==8777==  Address 0xffffffff00000c08 is not stack'd, malloc'd or (recently) free'd
==8777==
==8777==
==8777== Process terminating with default action of signal 11 (SIGSEGV)
==8777==  Bad permissions for mapped region at address 0xFFFFFFFF00000C08
==8777==    at 0xFFFFFFFF00000C08: ???
==8777==    by 0xAC: ???
==8777==    by 0x1190FF3F: ???
==8777==    by 0x12C2D9F: ??? (in /home/vladimir/.phpenv/versions/7.1.0-nts-debug/bin/php)
```

**Comment by Jurigag** [comment]
`Bad permissions for mapped region at address 0xFFFFFFFF00000C08` i guess on clang this permissions are ignored and causing it to work or what?

**Comment by sjinks** [comment]
You won't believe how stupid the error is:

```c
                        if (likely(param_count) <= 10) {
                                params_ptr = static_params;
                        } else {
                                params_arr = emalloc(param_count * sizeof(zval*));
                                params_ptr = params_arr;
                        }
```

This line:
```c
likely(param_count) <= 10
```

`likely(param_count)` is `__builtin_expect(!!(param_count), 1)` and it evaluates to 1. Then the first branch is taken, as `1 <= 10`.

**Pull request closed without merging**